components/brave_wallet/browser/internal/hd_key_ed25519_slip23.h

```diff
@@ -0,0 +1,70 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_ED25519_SLIP23_H_
+#define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_INTERNAL_HD_KEY_ED25519_SLIP23_H_
+
+#include <memory>
+#include <string_view>
+
+#include "base/containers/span.h"
+#include "base/types/pass_key.h"
+#include "brave/components/brave_wallet/browser/internal/hd_key_common.h"
+
+namespace brave_wallet {
+
+inline constexpr size_t kSlip23ScalarSize = 32;
+inline constexpr size_t kSlip23DerivationScalarSize = 28;
+inline constexpr size_t kSlip23PrefixSize = 32;
+inline constexpr size_t kSlip23ChainCodeSize = 32;
+
+// This class implements EdDSA over ed25519 functionality as SLIP-0023
+// https://github.com/satoshilabs/slips/blob/master/slip-0023.md
+class HDKeyEd25519Slip23 {
+ public:
+  using PassKey = base::PassKey<HDKeyEd25519Slip23>;
+
+  HDKeyEd25519Slip23();
```

Comment:
I may be wrong but it looks like this default constructor is not used, and it feels like it shouldn't be here?